### PR TITLE
CB-174: Changed navbar-form for uncollapsed navbar

### PR DESCRIPTION
**Pull request by ps2611**
Solved issue CB-174. When min-width: grid-float breakpoint satisfies, user has to first select search type entity and then search.It reduces the number of boxes to one at all times.

### CODE REVIEW

**Comment by gentlecat** [comment]
@brainzbot, ok to test.

File: critiquebrainz/frontend/templates/navbar.html [hunk 1]
```diff
@@ -63,3 +64,57 @@
...
+
+var searchBox = document.getElementById("navbar-search");
+var typeSelect = document.getElementById("type-selector");
+var mediaQquery = window.matchMedia("(min-width: 992px)");
+// Call listener function at run time
+driverFunction(mediaQquery);    
+// Attach listener function on state changes                        
+mediaQquery.addListener(driverFunction);             
+
+</script>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/navbar.html]: New scripts should be built separately. Files go into the `/critiquebrainz/frontend/static/scripts` directory.

File: critiquebrainz/frontend/templates/navbar.html [hunk 2]
```diff
@@ -52,6 +52,7 @@
         </div>
         <div class="form-group">
           <select id="type-selector" name="type" class="form-control input-sm">
+            <option value="" selected disabled>{{ _('Select Search Type') }}</option>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/navbar.html]: Capitalisation should be consistent with the rest of the items:

> Select search type

File: critiquebrainz/frontend/templates/navbar.html [hunk 3]
```diff
@@ -63,3 +64,57 @@
...
+}
+
+function focusout() {
+     typeSelect.style.display = 'inline-block';
+     searchBox.style.display = 'none';
+}
+
+var searchBox = document.getElementById("navbar-search");
+var typeSelect = document.getElementById("type-selector");
+var mediaQquery = window.matchMedia("(min-width: 992px)");
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/navbar.html]: Is `Qquery` a typo?

File: critiquebrainz/frontend/templates/navbar.html [hunk 4]
```diff
@@ -63,3 +64,57 @@
...
+    typeSelect.setAttribute("onchange", "select()");
+  }
+  else {
+    searchBox.style.display = 'inline-block';
+    searchBox.removeAttribute("onblur");
+    typeSelect.removeAttribute("onchange");
+  }
+}
+
+function select() {
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/navbar.html]: Can the name of this function be more descriptive? Can you provide some documentation in the comment that tells what this function is for?

Comment (gentlecat) [inline critiquebrainz/frontend/templates/navbar.html]: Same applies to the other functions.

File: critiquebrainz/frontend/templates/navbar.html (hunk 2 above)
Comment (ps2611) [inline critiquebrainz/frontend/templates/navbar.html]: Done that and took care for other strings as well

File: critiquebrainz/frontend/templates/navbar.html (hunk 4 above)
Comment (ps2611) [inline critiquebrainz/frontend/templates/navbar.html]: Changed name of one of the functions and provided proper description

File: critiquebrainz/frontend/templates/navbar.html (hunk 3 above)
Comment (ps2611) [inline critiquebrainz/frontend/templates/navbar.html]: Yes. Sorry for that

File: critiquebrainz/frontend/templates/navbar.html (hunk 1 above)
Comment (ps2611) [inline critiquebrainz/frontend/templates/navbar.html]: The sole reason to include script here is  to use translatable strings for template output.
(Search field placeholder values)

Comment (gentlecat) [inline critiquebrainz/frontend/templates/navbar.html]: There should still be a way to translate strings in JavaScript. Maybe see how MusicBrainz does it?

**Comment by gentlecat** [comment]
Looks like there are 3 other commits made by others that shouldn't be here.

**Comment by ps2611** [comment]
Yes. You are right. Sorry for this mess.

**Comment by ps2611** [comment]
And I can't seem to understand why it doesn't work on your setup. It works perfectly fine for me.
pjimage

**Comment by gentlecat** [comment]
Here's what I see currently:
screenshot 2017-03-24 02 44 32
Maybe I'm doing something wrong. Can someone else check?

**Comment by ferbncode** [comment]
Changes on input box work but the search does not work when done via the navbar. 

Second Screenshot:

**Comment by gentlecat** [comment]
I'm going to close this since there are clearly some issues, which need to be fixed. Feel free to reopen the PR if you get back to it.

**Pull request closed without merging**